src/traces/streamtube/calc.js

```diff
@@ -21,6 +21,10 @@ function calc(gd, trace) {
         trace.w.length
     );
 
+    ['u', 'v', 'w', 'x', 'y', 'z'].forEach(function(e) {
+        trace['_' + e] = filter(trace[e], trace._len);
+    });
```

Comment:
Same here. Please rewrite this as six almost-identical lines.

Reply:
Done in https://github.com/plotly/plotly.js/pull/4431/commits/c82da07617b8cac581eea4a4cc291a710683d320.